### PR TITLE
added ReferenceCount and ReferenceByIndex functions to script adapter

### PR DESCRIPTION
**Pull request by fireundubh**
- added ReferenceCount and ReferenceByIndex functions to script adapter
- renamed Reference properties for consistency with ReferencedBy properties (per discussion following the implementation of these properties)

### CODE REVIEW

File: Core/wbImplementation.pas [hunk 1]
```diff
@@ -1181,9 +1181,9 @@   TwbMainRecord = class(TwbRecord, IwbMainRecord, IwbMainRecordInternal, IwbMain
     procedure SortReferencedBy;
     function GetReferencedBy(aIndex: Integer): IwbMainRecord;
     function GetReferencedByCount: Integer;
-    function GetReferences(aIndex: Integer): IwbMainRecord;
-    function GetReferencesCount: Integer;
-    function GetExternalReferencesCount: Integer;
+    function GetReferenceBy(aIndex: Integer): IwbMainRecord;
```
Comment (ElminsterAU) [inline Core/wbImplementation.pas]: function GetReference(aIndex: Integer): IwbMainRecord;

File: Core/wbInterface.pas [hunk 2]
```diff
@@ -1726,12 +1726,12 @@   EwbSkipLoad = class(EAbort);
     property ReferencedByCount: Integer
       read GetReferencedByCount;
 
-    property Referenceds[aIndex: Integer]: IwbMainRecord
-      read GetReferences;
-    property ReferencedsCount: Integer
-      read GetReferencesCount;
-    property ExternalReferencedsCount: Integer
-      read GetExternalReferencesCount;
+    property ReferenceBy[aIndex: Integer]: IwbMainRecord
```
Comment (ElminsterAU) [inline Core/wbInterface.pas]: property References[aIndex: Integer]: IwbMainRecord

File: Core/wbImplementation.pas [hunk 3]
```diff
@@ -1181,9 +1181,9 @@   TwbMainRecord = class(TwbRecord, IwbMainRecord, IwbMainRecordInternal, IwbMain
...
+    function GetReferenceCount: Integer;
```
Comment (ElminsterAU) [inline Core/wbImplementation.pas]: function GetReferencesCount: Integer;

File: Core/wbImplementation.pas [hunk 4]
```diff
@@ -1181,9 +1181,9 @@   TwbMainRecord = class(TwbRecord, IwbMainRecord, IwbMainRecordInternal, IwbMain
...
+    function GetExternalReferenceCount: Integer;
```
Comment (ElminsterAU) [inline Core/wbImplementation.pas]: function GetExternalReferencesCount: Integer;

File: Core/wbInterface.pas [hunk 5]
```diff
@@ -1726,12 +1726,12 @@   EwbSkipLoad = class(EAbort);
...
+      read GetReferenceBy;
+    property ReferenceCount: Integer
```
Comment (ElminsterAU) [inline Core/wbInterface.pas]: property ReferencesCount: Integer

File: Core/wbInterface.pas [hunk 6]
```diff
@@ -1726,12 +1726,12 @@   EwbSkipLoad = class(EAbort);
...
+      read GetReferenceCount;
+    property ExternalReferenceCount: Integer
```
Comment (ElminsterAU) [inline Core/wbInterface.pas]: property ExternalReferencesCount: Integer

**Review by ElminsterAU — CHANGES_REQUESTED** [review]

**Comment by fireundubh** [comment]
Okay, I think I resolved all your requested changes. Please verify.

**Pull request merged**